python/jiminy_py/src/jiminy_py/simulator.py

```diff
@@ -357,36 +397,50 @@ def reset(self, remove_all_forces: bool = False) -> None:
         self.engine.reset(False, remove_all_forces)
 
     def start(self,
-              q_init: np.ndarray,
-              v_init: np.ndarray,
-              a_init: Optional[np.ndarray] = None,
-              is_state_theoretical: bool = False) -> None:
+              q_init: Union[np.ndarray, Dict[str, np.ndarray]],
+              v_init: Union[np.ndarray, Dict[str, np.ndarray]],
+              a_init: Optional[Union[np.ndarray, Dict[str, np.ndarray]]] = None,
+              is_state_theoretical: Optional[bool] = None) -> None:
         """Initialize a simulation, starting from (q_init, v_init) at t=0.
 
-        :param q_init: Initial configuration.
-        :param v_init: Initial velocity.
-        :param a_init: Initial acceleration. It is only used by acceleration
-                       dependent sensors and controllers, such as IMU and force
-                       sensors.
-        :param is_state_theoretical: Whether the initial state is associated
+        :param q_init: Initial configuration (by robot if it is a dictionnary).
+        :param v_init: Initial velocity (by robot if it is a dictionnary).
+        :param a_init: Initial acceleration (by robot if it is a dictionnary).
+                       It is only used by acceleration dependent sensors and
+                       controllers, such as IMU and force sensors.
+        :param is_state_theoretical: In single robot simulations, whether the initial state is associated
                                      with the actual or theoretical model of
                                      the robot.
         """
         # Call base implementation
-        self.engine.start(q_init, v_init, a_init, is_state_theoretical)
+        if len(self.engine.robots) == 1:
+            if isinstance(q_init, dict):
+                q_init = q_init[list(q_init.keys())[0]]
+                v_init = v_init[list(v_init.keys())[0]]
+                if a_init:
+                    a_init = a_init[list(a_init.keys())[0]]
+            if is_state_theoretical is None:
+                is_state_theoretical = False
+            self.engine.start(q_init, v_init, a_init, is_state_theoretical)
+        else:
+            if is_state_theoretical is not None:
+                raise Exception(
+                    "`is_state_theoretical` can only be precised in single "
+                    "robot simulations.")
+            self.engine.start(q_init, v_init, a_init)
 
         # Share the external force buffer of the viewer with the engine.
         # Note that the force vector must be converted to pain list to avoid
         # copy with external sub-vector.
-        if self.viewer is not None:
-            self.viewer.f_external = [*self.robot_state.f_external][1:]
+        for i, viewer in enumerate(self._viewers):
```

Comment:
zip would be better, as it also may sure that it will have the appropriate length, ie the min between the number of elements of `robot_states` and `self._viewers`:
```python
for robot_state, viewer in zip(self.robot_states, self._viewers):
    ....
```